files/image_config/procdockerstats/procdockerstats

```diff
@@ -0,0 +1,155 @@
+# !/usr/bin/env python
+'''
+This code is for a specific daemon, process-docker.
+Which sends process and docker CPU/memory utilization data to DB every 2 mins.
+'''
+
+import sys, errno
+import os
+import time
+import syslog
+import signal
+import select
+import subprocess
+import re
+import swsssdk 
+import collections
+from datetime import datetime
+
+VERSION = '1.0'
+
+SYSLOG_IDENTIFIER = "process-docker"
+
+REDIS_HOSTIP = "127.0.0.1"
+
+# ========================== Syslog wrappers ==========================
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+# ========================== ProcessDocker class ==========================
+class ProcDockerStats:
+
+    def __init__(self):
+        self.swid = 0
+        self.running = False
+        self.handle = None
+        
+    def initialize(self):
+        self.state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+        self.state_db.connect("STATE_DB")
+        # wipe out all data before updating with new values
+        self.state_db.delete_all_by_pattern('STATE_DB', 'Docker_Stats|*')
+              
+    def run_commands(self, commands):
+        """
+        Given a list of shell commands, run them in order
+        Args:
+            commands: List of strings, each string is a shell command
+        """
+        for cmd in commands:
+            proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+            (stdout, stderr) = proc.communicate()
+
+            if proc.returncode != 0:
+                log_error("Error running command '{}'".format(cmd))
+            else:
+                lines = re.split("\n", stdout)
+                keys = re.split("   +", lines[0])
+                x = len(lines)
+                dict1 = dict()
+                dict_list = []
+                for i in range(1,x):
+                    values1 = re.split("   +", lines[i])
+                    dict1 =  dict(zip(keys, values1))
+                    dict_list.append(dict1)
+                return dict_list                                        
+                                     
+    def update_process_docker_commands(self):
```

Comment:
I think this should be split into two functions: one for updating process stats and one for updating Docker stats.

Reply:
agree. added separate methods